test/plugin-barcode-scan.js

```diff
@@ -140,10 +139,142 @@ describe('Barcode scan', function() {
                 code: result
             }
         })));
+
+        document.dispatchEvent(new KeyboardEvent('keydown', {'key': 'Enter', 'code': 'Enter'}));
         setTimeout(function() {
             document.dispatchEvent(new KeyboardEvent('keydown', {'key': 'Enter', 'code': 'Enter'}));
             assert(onKeyDownCallback.called);
             done();
         }, 100);
     });
+
+    it('should move focus only after two chars are picked up in time based scan', function() {
+        const doc = window.document;
+        var keys = ['8', 'x'];
+        const input = doc.createElement("input");
+        input.setAttribute('id', 'focus-scan-barcode');
+        doc.body.appendChild(input);
+        input.focus();
+
+        keys.forEach(function(c) {
+            var key;
+            var code;
+            if( Object.prototype.toString.call(c) === '[object Object]' ) {
+                key = c.key;
+                code = c.code;
+            } else {
+                key = c;
+                code = `Key${c.toUpperCase()}`;
+            }
+            var event = new KeyboardEvent('keydown', { 'key': key, 'code': code });
+            doc.dispatchEvent(event);
+        });
+        assert.equal(doc.activeElement.id, 'focus-scan-barcode');
+        doc.dispatchEvent(new KeyboardEvent('keydown', { 'key': 'r', 'code': 'KeyR' }));
+        assert.notEqual(doc.activeElement.id, 'focus-scan-barcode');
+        document.dispatchEvent(new KeyboardEvent('keydown', {'key': 'Enter', 'code': 'Enter'}));
+    });
+
+    it('should prevent default and stop propagation only after two chars are picked up in time based scan', function() {
+        const doc = window.document;
+        var keys = ['8', 'x'];
```

Comment:
Unfortunately we use really old version of babel and weback and it would be time consuming and risky to switch to const (especially because this lib is used on web).